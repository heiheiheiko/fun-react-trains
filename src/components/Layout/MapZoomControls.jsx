import React, { Component } from 'react';
import { connect } from "react-redux";
import PropTypes from 'prop-types';
import { updateMapZoom } from '../../redux/actions';

class MapZoomControls extends Component {
  onMinusClick() {
    this.props.updateMapZoom(this.props.mapZoom - 1);
  }
  
  onPlusClick() {
    this.props.updateMapZoom(this.props.mapZoom + 1);
  }

  render() {
    return (
      <div className="f-row f-ccenter">
        <a className="button is-outlined is-primary has-text-weight-semibold" onClick={() => this.onMinusClick()}>-</a>
        <div className="is-size-4 mx-small has-text-weight-semibold"> {this.props.mapZoom} </div>
        <a className="button is-outlined is-primary has-text-weight-semibold" onClick={() => this.onPlusClick()}>+</a>
      </div>
     );
  }
}

MapZoomControls.propTypes = {
  mapZoom: PropTypes.number,
  updateMapZoom: PropTypes.func,
};

const mapState = state => ({ 
  mapZoom: state.mapZoom,
});

const mapDispatch = dispatch => ({
  updateMapZoom: mapZoom => dispatch(updateMapZoom(mapZoom)),
});

const ConnectedComponent = connect(mapState, mapDispatch)(MapZoomControls);
export default ConnectedComponent;
