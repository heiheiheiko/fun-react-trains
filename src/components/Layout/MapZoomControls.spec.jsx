import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer'
import SpecContainer from '../../../__specs__/helpers/SpecReduxIntelContainer'

import MapZoomControls from './MapZoomControls';

describe('<MapZoomControls>', () => {
  let componentContainer;

  beforeEach(() => {
    componentContainer = <SpecContainer><MapZoomControls /></SpecContainer>;
  })

  it('could be renderd', () => {
    expect(mount(componentContainer).find(MapZoomControls).length).toEqual(1)
  });

  it('matched the snapshot', () => {
    const renderedValue =  renderer.create(componentContainer).toJSON()
    expect(renderedValue).toMatchSnapshot();
  });
});