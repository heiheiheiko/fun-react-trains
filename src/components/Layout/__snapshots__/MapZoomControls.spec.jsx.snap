// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapZoomControls> matched the snapshot 1`] = `
<div>
   
  <div
    className="f-row f-ccenter"
  >
    <a
      className="button is-outlined is-primary has-text-weight-semibold"
      onClick={[Function]}
    >
      -
    </a>
    <div
      className="is-size-4 mx-small has-text-weight-semibold"
    >
       
      5
       
    </div>
    <a
      className="button is-outlined is-primary has-text-weight-semibold"
      onClick={[Function]}
    >
      +
    </a>
  </div>
   
</div>
`;
